packages/mask/src/plugins/Referral/SNSAdaptor/AdjustFarmRewards.tsx

```diff
@@ -0,0 +1,379 @@
+import { useCallback, useState } from 'react'
+import { useAsync } from 'react-use'
+import { formatUnits } from '@ethersproject/units'
+import { Chip, Grid, InputAdornment, TextField, Typography } from '@mui/material'
+import { makeStyles, useCustomSnackbar } from '@masknet/theme'
+import { Box } from '@mui/system'
+import {
+    EthereumTokenType,
+    formatBalance,
+    useAccount,
+    useChainId,
+    useFungibleTokenBalance,
+    useWeb3,
+    FungibleTokenDetailed,
+} from '@masknet/web3-shared-evm'
+
+import { AdjustFarmRewardsInterface, TransactionStatus, PagesType, TabsReferralFarms } from '../types'
+import { useI18N } from '../../../utils'
+import { EthereumChainBoundary } from '../../../web3/UI/EthereumChainBoundary'
+import ActionButton from '../../../extension/options-page/DashboardComponents/ActionButton'
+import { useRequiredChainId } from '../hooks/useRequiredChainId'
+import { roundValue } from '../helpers'
+import { APR, ATTRACE_FEE_PERCENT } from '../constants'
+import { referralFarmService, farmsService } from '../Worker/services'
+
+import { FarmTokenDetailed } from './shared-ui/FarmTokenDetailed'
+
+import { useSharedStyles } from './styles'
+
+const useStyles = makeStyles()((theme) => ({
+    valueCol: {
+        display: 'flex',
+        flexDirection: 'column',
+        '& span': { fontWeight: 600, marginTop: '4px' },
+    },
+    balance: {
+        whiteSpace: 'nowrap',
+        textOverflow: 'ellipsis',
+        overflow: 'hidden',
+        maxWidth: '80%',
+        fontSize: 12,
+        top: theme.spacing(0.5),
+    },
+    textField: {
+        '& input[type=number]': {
+            height: 30,
+            '-moz-appearance': 'textfield',
+        },
+        '& input[type=number]::-webkit-outer-spin-button': {
+            '-webkit-appearance': 'none',
+            margin: 0,
+        },
+        '& input[type=number]::-webkit-inner-spin-button': {
+            '-webkit-appearance': 'none',
+            margin: 0,
+        },
+    },
+}))
+
+export function AdjustFarmRewards(props: AdjustFarmRewardsInterface) {
+    const { farm, rewardToken, referredToken } = props
+    localStorage.setItem('adjustFarmRewardsData', JSON.stringify({ farm, rewardToken, referredToken }))
```

Comment:
Add `useEffect` for it.

Reply:
DON'T set data inside `localStorage`. It will leak to the SNS website.